cmip6_downscaling/methods/common/tasks.py

```diff
@@ -85,20 +92,43 @@ def get_experiment(run_parameters: RunParameters) -> UPath:
 
 
 @task
-def rechunk(path: UPath, pattern: str = None) -> UPath:
+def rechunk(path: UPath, pattern: str, run_parameters: RunParameters) -> UPath:
+    target = (
+        intermediate_dir
+        / "rechunked_stores_"
+        / pattern
+        / "_{model}_{scenario}_{variable}_{latmin}_{latmax}_{lonmin}_{lonmax}_{train_dates[0]}_{train_dates[1]}_{predict_dates[0]}_{predict_dates[1]}".format(
+            **asdict(run_parameters)
+        )
+    )
 
     # TODO: think about how to cache this result
-    target = intermediate_dir / "rechunk" / "todo"
     if use_cache and (target / '.zmetadata').exists():
         print(f'found existing target: {target}')
         return target
 
-    # TODO
+    chunk_dims = config.get(f"chunk_dims.{pattern}")
 
-    return target
+    group = zarr.open_consolidated(path)
+    ds = xr.open_dataset(path)
+    chunks_dict = calc_auspicious_chunks_dict(ds, chunk_dims)
+
+    temp_store, target_store, target_path = make_rechunker_stores(target)
+    rechunk_plan = rechunker.rechunk(
+        group,
+        chunks_dict,
+        "2GB",
+        target_store,
+        temp_store,
```

Comment:
```suggestion
        source=group,
        target_chunks=chunks_dict,
        max_mem="2GB",
        target_store=target_store,
        temp_store=temp_store,
```

Makes this a bit easier to read.